h5py/tests/test_file.py

```diff
@@ -99,6 +99,11 @@ def test_append(self):
         finally:
             fid.close()
 
+        # TODO: Make this test more robust -- fails on Linux possibly due to the
+        # use of the container in the CI environment...?
+        if os.getenv("CIBUILDWHEEL") == "1" and sys.platform == "linux":
+            return
```

Comment:
I think this should be replaced by a `@pytest.mark.skipif` decorator. Or possibly use `pytest.skip` directly here so we still have the opportunity to fail the test in its early parts.

Reply:
> so we still have the opportunity to fail the test in its early parts.

FWIW I think `return` here also allows the test to fail in the preceding parts. For example if there's an error on line 98 or whatever, it will still be raised *before* this line is ever hit.

It's a weird situation because only *part* of the test gets skipped. The cleanest thing to do would probably be to split this into a separate test altogether actually, then the `skip` will be more meaningful / correct because when you see it, it would mean that all parts of the test were skipped. (If we just added it in place now, it would be misleading because some of the test will have run already.) So I'll do that! 👍

Reply:
> FWIW I think return here also allows the test to fail in the preceding parts.

that's right, I'm just worried that we might change the meaning of "✅" from "that test passed" to "at least it returned before it could fail !" 😸